server/internal/interface/grpc/handlers/arkservice.go

```diff
@@ -145,11 +201,36 @@ func (h *handler) RegisterPayment(ctx context.Context, req *arkv1.RegisterPaymen
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
+	var cosigner *secp256k1.PublicKey
+
+	if h.svc.IsCovenantLess() {
+		pubkey := req.GetEphemeralPubkey()
+		if pubkey == "" {
+			return nil, status.Error(codes.InvalidArgument, "missing cosigner pubkey")
+		}
+
+		buf, err := hex.DecodeString(pubkey)
+		if err != nil {
+			return nil, status.Error(codes.InvalidArgument, "invalid cosigner pubkey (expect hex)")
+		}
+
+		cosigner, err = secp256k1.ParsePubKey(buf)
+		if err != nil {
+			return nil, status.Error(codes.InvalidArgument, "invalid cosigner pubkey")
+		}
+	}
```

Comment:
this is not really needed. We can just parse and the key if it's not nil, then at app level uses or ignores is depending on its version (the `missing cosigner pubkey` error message would be returned by the app svc):
```suggestion
	if pubkey := req.GetEphemeralPubkey(); len(pubkey) > 0 {
		buf, err := hex.DecodeString(pubkey)
		if err != nil {
			return nil, status.Error(codes.InvalidArgument, "invalid cosigner pubkey (expect hex)")
		}

		cosigner, err = secp256k1.ParsePubKey(buf)
		if err != nil {
			return nil, status.Error(codes.InvalidArgument, "invalid cosigner pubkey")
		}
	}
```